test/Bitmaps/BitmapFile.test.cpp

```diff
@@ -1,17 +1,60 @@
 #include "../src/Bitmaps/BitmapFile.h"
+#include "XFile.h"
 #include <gtest/gtest.h>
 #include <stdexcept>
 
+void RoundTripSub(uint16_t bitCount, int32_t width, int32_t height)
+{
+	const std::string filename("Sprites/data/BitmapTest.bmp");
+
+	BitmapFile bitmapFile = BitmapFile::CreateDefaultIndexed(bitCount, width, height);
+	BitmapFile bitmapFile2;
+
+	EXPECT_NO_THROW(BitmapFile::WriteIndexed(filename, bitmapFile));
+	EXPECT_NO_THROW(bitmapFile2 = BitmapFile::ReadIndexed(filename));
+	EXPECT_EQ(bitmapFile, bitmapFile2);
```

Comment:
I would probably have used a memory stream here, rather than an actual disk file. Less cleanup, faster tests, and more robust in case files liter the filesystem or there are permission issues.

One downside is the filename variant of the Read/Write methods tends to wrap the raw stream methods, so technically you'd be testing a bit less. That means more tests to exercise the wrappers. At least, assuming you care about 100% code coverage. Might be sensible to aim for more of an 80% solution.

Anyway, curious what you think on the matter. I could see arguments for both ways.